src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -66,16 +69,26 @@
     @Value("${jhipster.clientApp.name}")
     private String applicationName;
 
+    @Value("${artemis.external-user-management:true}")
+    private boolean EXTERNAL_USER_MANAGEMENT;
+
     private final UserRepository userRepository;
 
     private final UserService userService;
 
+    private final Optional<ContinuousIntegrationService> continuousIntegrationService;
```

Comment:
I am not sure if it is a good idea to have these dependencies in the Resource. I guess we should move all this logic into the UserService

Reply:
Yes, this was a bad design choice. I moved the relevant code to the service